product_customer_code_invoice/models/account_invoice.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2017 Agile Business Group sagl
+#     (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.multi
+    def _get_product_customer_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for line in self.filtered(lambda il: il.product_id.supplier_ids):
+            partner = line.partner_id
+            product = line.product_id
+            if product.supplier_ids and partner:
```

Comment:
IMHO this check is not necessary as the partner is mandatory and you are filtering the lines with products with supplierinfo.